pygame/classify_capture.py

```diff
@@ -14,16 +14,61 @@
 
 """A demo to classify Pygame camera stream."""
 import argparse
-import os
-import io
-import time
+import collections
 from collections import deque
+import io
 import numpy as np
+import operator
+import os
 import pygame
 import pygame.camera
 from pygame.locals import *
+import tflite_runtime.interpreter as tflite
+import time
+
+Class = collections.namedtuple('Class', ['id', 'score'])
+EDGETPU_SHARED_LIB = 'libedgetpu.so.1'
+
+def make_interpreter(model_file):
+    model_file, *device = model_file.split('@')
+    return tflite.Interpreter(
+      model_path=model_file,
+      experimental_delegates=[
+          tflite.load_delegate(EDGETPU_SHARED_LIB,
+                               {'device': device[0]} if device else {})
+      ])
+
+def input_size(interpreter):
+    """Returns input image size as (width, height, channels) tuple."""
+    _, height, width, channels = interpreter.get_input_details()[0]['shape']
+    return width, height, channels
+
+def input_tensor(interpreter):
+    """Returns input tensor view as numpy array of shape (height, width, 3)."""
+    tensor_index = interpreter.get_input_details()[0]['index']
+    return interpreter.tensor(tensor_index)()[0]
+
+def output_tensor(interpreter):
+    """Returns dequantized output tensor."""
+    output_details = interpreter.get_output_details()[0]
+    output_data = np.squeeze(interpreter.tensor(output_details['index'])())
+    scale, zero_point = output_details['quantization']
+    return scale * (output_data - zero_point)
+
+def set_interpreter(interpreter, data):
```

Comment:
This is an odd name since it invokes the interpreter. Why not "invoke_interpreter" ?
Honestly i'd just get rid of this helper entirely. The reshape really belongs with the buffer retriever step, not with the invoker step and interpreter.invoke() is already a clean call.